//handling nesting executions
const timeToWait = 2000 //2 seconds - to simulate a http request
const someTask = async () => {
    return new Promise((resolve, reject) => {
        setTimeout(() => resolve('It works'), timeToWait);
    })
}

console.log('\n')
someTask()
    .then(console.log('1ยบ nested execution done!\n'))
    .then(console.log('2ยบ nested execution done!\n'))
    .then(() => {
        console.log('3ยบ nested execution will throw an error!\n')
        throw new Error('Oops! something went wrong in 3ยบ nested execution.')
    })        
    .catch(error => console.log(error))

